### ISSUE TITLE
Option to gather UTXO's (either automatically, or by clicking a button)

### ISSUE THREAD

**Issue by BlakeBrown** [Berry-Pool/nami-wallet#134]
I've been brute testing how many UTXO's Nami can handle by using it as our marketplace wallet 😛 

We've now gathered 1200 UTXO's which results in "Transaction not possible" warnings when trying to consume all of them at once.

It'd be nice to have an option in Nami to gather UTXOs, or at least reduce them down to a reasonable set (10-20). Perhaps this could even be done automatically once the # of UTXOs exceeds a certain size? I can see more users running into this problem in the future, for example popular NFT creators will likely receive 1000's of UTXOs to their wallets in the form of royalties. 

<img src="https://user-images.githubusercontent.com/6283620/145688172-0545aad1-b2d3-45f2-bf85-14aacf64ca4e.png" width="250">

Thank you so much for making this incredible wallet and pushing the ecosystem forwards. I wish you could see the adoption of users we've had in the past few days who are thrilled they can easily interact with smart contracts on Cardano using Nami. WAGMI 😄

**Comment by pascallapointe** [comment]
Hi Blake, thanks for the comment!

This is a problem with many angles to consider.

Actually Nami do reduce the UTxO set every transaction a user do using Nami. So basically, the way it work is that every time a user do a transaction with Nami, the wallet try to generate a "change" UTxO equivalent to the amount of the transaction.
To match the transaction amount, the wallet order UTxO from the smallest ADA amount to the largest, and merge small UTxO.

This behaviour is actually limited to 20 UTxO to stay fee efficient, if the transaction required over 20 UTxO, the wallet won't try to improve the UTxO set as this would only make the transaction fees higher. The improvement will be done on a more optimal transaction.

We could add a feature to consolidate a wallet for NFT makers, but selling NFT is a really specific use case. The majority of user would be confused about this kind of feature as they do not understand the UTxO model and we would need to keep explaining why UTxO optimization is even a thing.
The consolidation of a wallet can be done through a self transaction (sending to your own address). Also, adding this kind of feature would be a challenge in itself as the feature would need to be able to analyze the set and determine what UTxO need to be optimized and where to stop.
Basically, adding a "send all/Send max" feature would serve the same purpose. (This feature is planned but more complex to implement that I first thought)

The Nami CoinSelection algorithm have some known flaw at the moment, I will review it in the upcoming month and try to nail the sweet spot between optimization/Transaction fees.

**Comment by BlakeBrown** [comment]
Hey @pascallapointe really appreciate you sharing your thoughts! 😄 Totally agreed, it is a really complex and interesting problem, but one we'll need to solve to grow Cardano.
